usr/share/rear/restore/default/900_create_missing_directories.sh

```diff
@@ -1,38 +1,49 @@
 #
-# some backup SW doesn't restore mountpoints properly
+# Create missing directories.
+# For background information and reasoning see
+# https://github.com/rear/rear/issues/1455#issuecomment-324904017
+# that reads:
+#   Typical backup software assumes that the restore always happens
+#   on top of a system that was installed the traditional way,
+#   so that for the backup software it is safe to assume that
+#   those directories are present. From the perspective
+#   of the backup software there is therefore no need
+#   to include such directories in a full backup.
+#   Bottom line is, it is the duty of ReaR to make sure
+#   that these special directories really do exist.
 #
-#
-# create missing directories
-pushd $TARGET_FS_ROOT >/dev/null
-if [[ -f "$VAR_DIR/recovery/mountpoint_permissions" ]] ; then
-    LogPrint "Restore the Mountpoints (with permissions) from $VAR_DIR/recovery/mountpoint_permissions"
-    while read _dir mode userid groupid
-    do
-        ! [[ -d "$_dir" ]] && mkdir -p $_dir
-        chmod $mode $_dir
-        chown $userid:$groupid $_dir
-    done < <(cat "$VAR_DIR/recovery/mountpoint_permissions")
-elif [[ -z "$MOUNTPOINTS_TO_RESTORE" ]] ; then
-    # keep this for backward compatibility
-    mkdir -p mnt proc run sys tmp dev/pts dev/shm
-else
-    # keep this for backward compatibility
-    LogPrint "Restore the Mountpoints $MOUNTPOINTS_TO_RESTORE".
-    mkdir -p $MOUNTPOINTS_TO_RESTORE
-    # ensure some important mountpoints will be restored:
-    for _dir in mnt proc sys tmp dev/pts dev/shm
-    do
-      ! [ -d "$_dir" ] && mkdir -p $_dir
-    done
+pushd $TARGET_FS_ROOT 1>/dev/null
+
+local tmp_directories="tmp var/tmp"
+# First of all some generic directories that are created in any case:
+for directory in mnt proc run sys dev/pts dev/shm $tmp_directories ; do
+    test -d "$directory" || mkdir $v -p $directory 1>&2
+done
+# Set permissions for 'tmp' directories (cf. issue #1455):
+for tmp_dir in $tmp_directories ; do
+    chmod $v 1777 $tmp_dir 1>&2
+done
+
+# Recreate mountpoints with permissions from the mountpoint_permissions file:
+local mountpoint_permissions_file="$VAR_DIR/recovery/mountpoint_permissions"
+if test -f "$mountpoint_permissions_file" ; then
```

Comment:
Why not simply add the hard coded directories and the `DIRECTORIES_TO_CREATE` and `MOUNTPOINTS_TO_RESTORE` stuff to the `$VAR_DIR/recovery/mountpoint_permissions` file and have only a single loop that creates directories and sets modes and owners? Less code → less things that can go wrong.

Reply:
If we rename _mountpoint_permissions_ to something more meaningful (e.g. directory_permissions), it could serve more general purpose and also solve problem with setting of permissions during recovery.

Reply:
I think I will enhance DIRECTORIES_TO_CREATE
as described in
https://github.com/rear/rear/issues/1455#issuecomment-324949532
so that I can add it in a compatible way to
the mountpoint_permissions file that contains
lines of the form
<pre>
path/to/directory permission userid groupid
</pre>
so that I do not need to guess permission userid groupid
in 900_create_missing_directories.sh for entries
the mountpoint_permissions file that do not have them.
E.g. for me mountpoint_permissions contains
<pre>
dev 755 root root
dev/hugepages 755 root root
dev/mqueue 1777 root root
dev/pts 755 root root
dev/shm 1777 root root
proc 555 root root
run 755 root root
run/user/0 700 root root
sys 555 root root
tmp/rear.tMwgPVZJc9ri54a/outputfs 777 root root
</pre>

Reply:
@gozora 
what mountpoint_permissions contains for me shows
that I really have to clean up this whole functionality
but this will happen next week.